src/yunohost/certificate.py

```diff
@@ -858,12 +858,40 @@ def _domain_is_accessible_through_HTTP(ip, domain):
         requests.head("http://" + ip, headers={"Host": domain}, timeout=10)
     except requests.exceptions.Timeout as e:
         logger.warning(m18n.n('certmanager_http_check_timeout', domain=domain, ip=ip))
-        return False
     except Exception as e:
         logger.debug("Couldn't reach domain '%s' by requesting this ip '%s' because: %s" % (domain, ip, e))
+    else:
+        return True
+
+    # fallback on external YunoHost service because http loopback might be
+    # broken (this is actually way more common than excepted and LE doesn't
+    # care about local loopback, it wants outside reachability)
+
+    try:
+        response = requests.post("https://http-check.yunohost.org/check/", data={"domain": domain})
```

Comment:
That's more a comment on the server side than the client side, but shouldn't we actually check that this is a yunohost instance rather than just if it answers to HTTP request ? I'm concerned about routers running a web server themselves (this is more and more the case) hence that would bring false-positives in that case.

More generally I'm wondering if this ain't a bit like reinventing sort of the ACME challenge ourselves (if we wanted to cover more complicated use case with several yunohost in reverse-proxy config and so on, we would be generating a random id and check if the server sees the correct id) and if we just shouldn't use the staging server of Let's Encrypt instead of running our own service with the (small, but existent) maintenance cost that comes with it :s

Reply:
> That's more a comment on the server side than the client side, but shouldn't we actually check that this is a yunohost instance rather than just if it answers to HTTP request ? (maybe checking for SSOwat headers ? idk) I'm concerned about routers running a web server themselves (this is more and more the case) hence that would bring false-positives in that case.

That would be the next level of check, that would indeed be better and: 

> we would be generating a random id and check if the server sees the correct id

Would also be the way I would do it. Right now I wanted to only do the next step to fix the most common bugs and escalate to a more advanced solution if needed (I really don't like huge PR that are hard to read).

Also because I don't have a lot of energy and motivation right now and it was already quite some work to do >_<

> More generally I'm wondering if this ain't a bit like reinventing sort of the ACME challenge ourselves [...] and if we just shouldn't use the staging server of Let's Encrypt instead of running our own service with the (small, but existent) maintenance cost that comes with it :s

On one hand yes, on the other hand I had in mind to use this new server in the future diagnostic tool to check if a domain is correctly configured so that would be more than just just redoing LE check but I understand your sentiment.

Reply:
>check if a domain is correctly configured so that would be more than just just redoing LE check but I understand your sentiment.

Well in fact yes, I was thinking that the whole ip/domain rate thing looks pretty neat and that would definitely be useful to check the ports (instead of the old ports.yunohost.org which i believe don't have any rate limiting thing and isn't 100% reliable :P)